### ISSUE TITLE
Add `'no-undef': 'off'` for typescript files

### ISSUE THREAD

**Issue by kidkkr** [iamturns/eslint-config-airbnb-typescript#50]
If `no-undef` is on, it will false positive on typescript interface export.
This is because of https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb-base/rules/variables.js#L27-L28

```ts
interface Foo { /* ... */ }
export default Foo // eslint no-undef error
```

**Comment by iamturns** [comment]
Thanks for reporting this @kidkkr !

Related issue in ESLint: https://github.com/typescript-eslint/typescript-eslint/issues/342
